snuba/clickhouse/native.py

```diff
@@ -127,6 +130,57 @@ def close(self) -> None:
             pass
 
 
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+def transform_nullable(
+    function: Callable[[T], R]
+) -> Callable[[Optional[T]], Optional[R]]:
+    def transform_column(value: Optional[T]) -> Optional[R]:
+        if value is None:
+            return value
+        else:
+            return function(value)
+
+    return transform_column
+
+
+def transform_date(value: date) -> str:
+    """
+    Convert a timezone-naive date object into an ISO 8601 formatted date and
+    time string respresentation.
+    """
+    return datetime(*value.timetuple()[:6]).replace(tzinfo=tz.tzutc()).isoformat()
+
+
+def transform_datetime(value: datetime) -> str:
+    """
+    Convert a timezone-naive datetime object into an ISO 8601 formatted date
+    and time string representation.
+    """
+    return value.replace(tzinfo=tz.tzutc()).isoformat()
+
+
+def transform_uuid(value: UUID) -> str:
+    """
+    Convert a UUID object into a string representation.
+    """
+    return str(value)
+
+
+transform_result = build_result_transformer(
+    {
+        "Date": transform_date,
+        "Nullable(Date)": transform_nullable(transform_date),
+        "DateTime": transform_datetime,
+        "Nullable(DateTime)": transform_nullable(transform_datetime),
+        "UUID": transform_uuid,
+        "Nullable(UUID)": transform_nullable(transform_uuid),
```

Comment:
I get we expect to run different transformations between http and native, but are you expecting the list of types (`Date`, `Nullable(Date)`, etc.) we want to transform to also be different or just the transformation function to apply? If the list of types has to be the same we should probably have it in the parent class.